log/logging.go

```diff
@@ -188,22 +214,123 @@ func SetTimestampFlags(flags int) {
 	DefaultLogger.SetTimestampFlags(flags)
 }
 
+type LoggerInterface interface {
+	Fatal(description string, keysAndValues ...interface{})
+	Error(description string, keysAndValues ...interface{})
+	Warn(description string, keysAndValues ...interface{})
+	Info(description string, keysAndValues ...interface{})
+	Debug(description string, keysAndValues ...interface{})
+	Trace(description string, keysAndValues ...interface{})
+	SetOutput(w io.Writer)
+	SetTimestampFlags(flags int)
+	SetField(name string, value interface{})
+}
+
+func NewLogger(format LogFormat, id string, staticKeysAndValues ...interface{}) LoggerInterface {
+	return newLoggerStruct(format, id, staticKeysAndValues...)
+}
+
+func newLoggerStruct(format LogFormat, id string, staticKeysAndValues ...interface{}) *Logger {
+	var prefix string
+	var flags int
+	staticArgs := make(map[string]string, 0)
+
+	format = SanitizeFormat(format)
+
+	if format == JsonFormat {
+		// Don't mess up the json by letting logger print these:
+		prefix = ""
+		flags = 0
+
+		// Instead put them into the staticArgs
+		if defaultPrefix != "" {
```

Comment:
I know we discussed this yesterday, but handling both a json parseable string or just a string seems more complicated than it needs to be.

I'd suggest making prefix just a string and then if we want more context, let's add that support later. In retrospect, I think we might have jumped the gun on adding this complexity before we need it.